pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/PinotLogicalTableResourceTest.java

```diff
@@ -129,90 +133,191 @@ public void testCreateUpdateDeleteLogicalTables(String logicalTableName, List<St
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
       fail("Logical Table POST request should have failed");
     } catch (IOException e) {
-      assertTrue(e.getMessage().contains("Reason: 'tableName' should not end with _OFFLINE or _REALTIME"),
+      assertTrue(e.getMessage()
+              .contains("Reason: 'refRealtimeTableName' should not be null or empty when realtime table exists"),
           e.getMessage());
     }
 
-    // Test logical table name can not be same as existing physical table name
-    logicalTableConfig =
-        getDummyLogicalTableConfig("test_table_1", physicalTableNamesWithType, BROKER_TENANT);
+    // Test ref offline table is present in the offline tables validation
+    logicalTableConfig = getDummyLogicalTableConfig(LOGICAL_TABLE_NAME, physicalTableNamesWithType, BROKER_TENANT);
+    logicalTableConfig.setRefOfflineTableName("random_table_OFFLINE");
     try {
-      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
       fail("Logical Table POST request should have failed");
     } catch (IOException e) {
-      assertTrue(e.getMessage().contains("Table name: test_table_1 already exists"), e.getMessage());
+      assertTrue(e.getMessage().contains("Reason: 'refOfflineTableName' should be one of the provided offline tables"),
+          e.getMessage());
     }
 
+    // Test ref realtime table is present in the realtime tables validation
+    logicalTableConfig = getDummyLogicalTableConfig(LOGICAL_TABLE_NAME, physicalTableNamesWithType, BROKER_TENANT);
+    logicalTableConfig.setRefRealtimeTableName("random_table_REALTIME");
+    try {
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(
+          e.getMessage().contains("Reason: 'refRealtimeTableName' should be one of the provided realtime tables"),
+          e.getMessage());
+    }
+  }
+
+  @Test
+  public void testLogicalTableBrokerTenantValidation()
+      throws IOException {
+    List<String> physicalTableNamesWithType = createHybridTables(List.of("test_table_1"));
+    LogicalTableConfig logicalTableConfig =
+        getDummyLogicalTableConfig(LOGICAL_TABLE_NAME, physicalTableNamesWithType, "InvalidTenant");
+    try {
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Reason: 'InvalidTenant' should be one of the existing broker tenants"),
```

Comment:
`expectedException` and `excepectedExceptionErrorMessageRegexp` is preferred if this only ONE negative test. https://howtodoinjava.com/testng/testng-expected-exception/

Reply:
This is a nit. Feel free to ignore

Reply:
Done.